server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/QuotaGetMethod.scala

```diff
@@ -0,0 +1,154 @@
+        description = s"The following properties [${invalidProperties.format()}] do not exist.",
+        methodCallId = invocation.invocation.methodCallId))
+    })
+      .map(InvocationWithContext(_, invocation.processingContext))
+      .onErrorResume { case e: Exception => handleRequestValidationErrors(e, invocation.invocation.methodCallId)
+        .map(errorInvocation => InvocationWithContext(errorInvocation, invocation.processingContext))
+      }
+  }
+
+  override def getRequest(mailboxSession: MailboxSession, invocation: Invocation): Either[Exception, QuotaGetRequest] =
+    QuotaSerializer.deserializeQuotaGetRequest(invocation.arguments.value) match {
+      case JsSuccess(quotaGetRequest, _) => Right(quotaGetRequest)
+      case errors: JsError => Left(new IllegalArgumentException(ResponseSerializer.serialize(errors).toString))
+    }
+
+  private def getQuotaGetResponse(quotaGetRequest: QuotaGetRequest, username: Username): SFlux[QuotaResponseGetResult] =
+    quotaGetRequest.ids match {
+      case None =>
+        jmapQuotaManagerWrapper.list(username)
+          .collectSeq()
+          .map(listJmapQuota => QuotaResponseGetResult(jmapQuotaSet = listJmapQuota.toSet))
+          .flatMapMany(result => SFlux.just(result))
+      case Some(ids) => SFlux.fromIterable(ids.value)
+        .flatMap(id => jmapQuotaManagerWrapper.get(username, id.id.value)
+          .map(jmapQuota => QuotaResponseGetResult(jmapQuotaSet = Set(jmapQuota)))
+          .switchIfEmpty(SMono.just(QuotaResponseGetResult(notFound = QuotaNotFound(Set(id))))))
+    }
+
+  private def handleRequestValidationErrors(exception: Exception, methodCallId: MethodCallId): SMono[Invocation] = exception match {
+    case _: MissingCapabilityException => SMono.just(Invocation.error(ErrorCode.UnknownMethod, methodCallId))
+    case e: IllegalArgumentException => SMono.just(Invocation.error(ErrorCode.InvalidArguments, e.getMessage, methodCallId))
+  }
+}
+
+case class JmapQuotaManagerWrapper(private var quotaManager: QuotaManager,
+                                   private var quotaRootResolver: UserQuotaRootResolver) {
+  def get(username: Username, quotaId: String): SFlux[JmapQuota] =
+    SMono.just(quotaRootResolver.forUser(username))
+      .filter(quotaRoot => quotaId.isEmpty || QuotaIdFactory.from(quotaRoot).value.equals(quotaId))
+      .flatMapMany(getJmapQuota)
+
+  def list(username: Username): SFlux[JmapQuota] =
+    SMono.just(quotaRootResolver.forUser(username))
+      .flatMapMany(getJmapQuota)
+
+  private def getJmapQuota(quotaRoot: QuotaRoot): SFlux[JmapQuota] =
+    SMono(quotaManager.getQuotasReactive(quotaRoot))
+      .flatMapIterable(quotas => convertFromJava(quotas, quotaRoot))
+
+  private def convertFromJava(quotas: QuotaManager.Quotas, quotaRoot: QuotaRoot): List[JmapQuota] = {
+    val quotaId = QuotaIdFactory.from(quotaRoot)
+    val messageQuota = quotas.getMessageQuota
+    val jmapMessageQuotas = ListBuffer[JmapQuota]()
```

Comment:
We should refrain from using mutable code. Please write instead nice functional code:

 -> Extract 1 subMethod for `Option[JmapQuota` for count,  1 subMethod for `Option[JmapQuota` for space
 -> List(countQuota(...), sizeQuota(...)).flatten
 
 The code will be way more readable.